apps/explorer/src/hooks/useNetworkManager.ts

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import * as Sentry from '@sentry/react';
+import { useEffect, useState } from 'react';
+
+import { growthbook } from '~/utils/growthbook';
+import { queryClient } from '~/utils/queryClient';
+
+export function useNetworkManager(network: string) {
+    const [currentNetwork, setCurrentNetwork] = useState(network);
+    const [previousNetwork, setPreviousNetwork] = useState(network);
+
+    if (network !== currentNetwork) {
+        setPreviousNetwork(currentNetwork);
```

Comment:
Instead of creating a `usePrevious` hook or implementing this functionality with a `ref`, I'm opting to use state not directly abstract the "usePrevious" logic into a separate hook 

(https://twitter.com/dan_abramov/status/1653422709508849664?s=20)